### PR TITLE
Adding the function to handle preloaders

### PR DESCRIPTION
**Pull request by jcgt96**
Regarding to the discussed issue, I found that was due to a pre loader that did not allow to click in the clickable elements. So I added a new function to handle this kind of elements with an explicit wait until the pre loader element disappears.

### CODE REVIEW

File: src/main/pages/base_page.py [hunk 1]
```diff
@@ -82,17 +82,31 @@ def switch_to_iframe(self, by_locator):
...
+
         all_windows = self.driver.window_handles
         for window in all_windows:
             if window != main_window:
                 self.driver.switch_to.window(window)
+                self.wait_for_element_to_disappear(by_locator)
                 break
 
-    def switch_to_tab(self, tab_index):
+    def switch_to_tab(self, tab_index, by_locator):
```
Comment (AlexAndradeNet) [inline src/main/pages/base_page.py]: Rename the function to express the intention to switch to a tab and wait for an element to disappear. Another thing is renaming the variable to understand that it is the locator for the element you want to wait to disappear.

File: src/main/pages/base_page.py [hunk 2]
```diff
@@ -82,17 +82,31 @@ def switch_to_iframe(self, by_locator):
...
         tabs = self.driver.window_handles
+
         self.driver.switch_to.window(tabs[tab_index])
+        self.wait_for_element_to_disappear(by_locator)
+
+    def back_to_main_tab(self):
```
Comment (AlexAndradeNet) [inline src/main/pages/base_page.py]: You can create a function for switching to a tab with this code: 

```python
        tabs = self.driver.window_handles
        self.driver.switch_to.window(tabs[tab_index])
```

A function to switch_to_main_tab: `switch_to_tab(0)`.
Finally, you can create a third one for the switch and wait for an element to disappear that calls `switch_to_tab(x)` and wait.

**Review by AlexAndradeNet — CHANGES_REQUESTED** [review]

File: src/main/pages/base_page.py (hunk 1 above)
Comment (jcgt96) [reply src/main/pages/base_page.py]: I considered to express the intention in the HomePage file where is called this function. Besides, I decides to call the function that waits until the preloader disappears out of the switch function and put it in the HomePage within the function that makes these two actions, switch and wait.

File: src/main/pages/base_page.py (hunk 2 above)
Comment (jcgt96) [reply src/main/pages/base_page.py]: With the change mentioned above, this would not be necessary

**Pull request merged**